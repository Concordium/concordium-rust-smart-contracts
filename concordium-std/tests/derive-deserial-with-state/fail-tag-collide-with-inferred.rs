//! Ensure `derive(DeserialWithState)` fails when 'tag' attribute are collides
//! with the default tag for another variant.
use concordium_std::*;

#[derive(DeserialWithState)]
#[concordium(repr(u8))]
#[concordium(state_parameter = "S")]
enum MyTaggedEnum<S> {
    One(StateSet<u32, S>),
    #[concordium(tag = 0)]
    Two,
}

fn main() {}
