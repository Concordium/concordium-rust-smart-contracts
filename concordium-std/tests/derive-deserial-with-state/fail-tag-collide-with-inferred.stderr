error: The tag of 'Two' collides with the tag of 'One'.
  --> tests/derive-deserial-with-state/fail-tag-collide-with-inferred.rs:10:18
   |
10 |     #[concordium(tag = 0)]
   |                  ^^^

error: The tag of 'One' collides with the tag of 'Two'.
 --> tests/derive-deserial-with-state/fail-tag-collide-with-inferred.rs:9:5
  |
9 |     One(StateSet<u32, S>),
  |     ^^^
